pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,233 @@
+
+    grad_method = None
+    """Gradient computation method."""
+
+    def __init__(self, UA, projectors, wires, do_queue=True, id=None):
+        if not isinstance(UA, qml.operation.Operator):
+            raise ValueError("Input block encoding must be an Operator")
+
+        self._hyperparameters = {
+            "UA": UA,
+            "projectors": projectors,
+        }
+
+        super().__init__(wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def compute_decomposition(
+        UA, projectors, wires, **hyperparameters
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+
+        The :class:`~.QSVT` is decomposed into alternating block encoding
+        and projector-controlled phase shift operators. This is defined by the following
+        equations, where :math:`U` is the block encoding operation and both :math:`\Pi_\phi` and
+        :math:`\tilde{\Pi}_\phi` are projector-controlled phase shifts with angle :math:`\phi`.
+        When the number of projector-controlled phase shifts, :math:`d`, is odd:
+
+        .. math:: U_{QSVT} = \tilde{\Pi}_{\phi_1}U\prod^{(d-1)/2}_{k=1}{\Pi}_{\phi_{2k}}U^{\dagger}\tilde{\Pi}_{\phi_{2k+1}}U,
+
+        and when :math:`d` is even:
+
+        .. math:: U_{QSVT} = \prod^{d/2}_{k=1}{\Pi}_{\phi_{2k-1}}U^{\dagger}\tilde{\Pi}_{\phi_{2k}}U.
+
+        .. seealso:: :meth:`~.QSVT.decomposition`.
+
+        Args:
+            UA (Operator): the block encoding circuit, specified as a :class:`~.Operator`
+                or quantum function.
+            projectors (list[Operator]): a list of projector-controlled phase
+                shift circuits that implement the desired polynomial.
+            wires (Iterable): wires that the template acts on
+
+        Returns:
+            list[.Operator]: decomposition of the operator
+        """
+
+        op_list = []
+        UA_adj = UA.__copy__()
+
+        for idx, op in enumerate(projectors[:-1]):
+            qml.apply(op)
+            op_list.append(op)
+
+            if idx % 2 == 0:
+                qml.apply(UA)
+                op_list.append(UA)
+
+            else:
+                op_list.append(adjoint(UA_adj))
+
+        qml.apply(projectors[-1])
+        op_list.append(projectors[-1])
+
+        return op_list
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        op_label = base_label or self.__class__.__name__
+        return op_label
+
+    def queue(self, context=QueuingManager):
+        context.remove(self._hyperparameters["UA"])
+        for op in self._hyperparameters["projectors"]:
+            context.remove(op)
+        context.append(self)
+        return self
+
+
+def _qsp_to_qsvt(angles):
+    r"""Converts qsp angles to qsvt angles."""
+    new_angles = copy.copy(angles)
+    new_angles[0] += 3 * np.pi / 4
+    new_angles[-1] -= np.pi / 4
+
+    for i, phi in enumerate(new_angles[1:-1]):
```

Comment:
Does this doesn't work?
```suggestion
    new_angles[1:-1] += np.pi / 2
```

Reply:
TypeError: 'float' object is not iterable.